src/algebra/pointwise.lean

```diff
@@ -599,27 +599,70 @@ section
 
 variables {α : Type*} {β : Type*}
 
-/-- A nonempty set in a module is scaled by zero to the singleton
-containing 0 in the module. -/
-lemma zero_smul_set [semiring α] [add_comm_monoid β] [module α β] {s : set β} (h : s.nonempty) :
+lemma zero_smul_set [has_zero α] [has_zero β] [smul_with_zero α β] {s : set β} (h : s.nonempty) :
```

Comment:
Maybe keep the docstring. I know it's not quite accurate any more, but I'd rather have a slightly lying docstring than none.

Reply:
I'd argue the docstring is worthless anyway, as it just reads out the statement symbol by symbol (and inaccurate at that, since the RHS is `0` not `{0}`).

Reply:
```suggestion
/-- A nonempty set in is scaled by zero to the singleton set containing 0. -/
lemma zero_smul_set [has_zero α] [has_zero β] [smul_with_zero α β] {s : set β} (h : s.nonempty) :
```